snuba/web/views.py

```diff
@@ -604,16 +604,14 @@ def _write_to_entity(*, entity: EntityType) -> RespTuple:
 
         return ("ok", 200, {"Content-Type": "text/plain"})
 
-    @application.route("/tests/<dataset:dataset>/insert", methods=["POST"])
-    def write(*, dataset: Dataset) -> RespTuple:
-        return _write_to_entity(entity=dataset.get_default_entity())
-
     @application.route("/tests/entities/<entity:entity>/insert", methods=["POST"])
     def write_to_entity(*, entity: EntityType) -> RespTuple:
         return _write_to_entity(entity=entity)
 
-    @application.route("/tests/<dataset:dataset>/eventstream", methods=["POST"])
-    def eventstream(*, dataset: Dataset) -> RespTuple:
+    @application.route(
+        "/tests/<dataset:dataset>/<entity:entity>/eventstream", methods=["POST"]
```

Comment:
What's the current status of this in Sentry? Have you updated this endpoint to pass entity everywhere?

Reply:
From what I understand, this is purely a test endpoint used in both Snuba and Sentry. The Snuba eventstream tests have been updated accordingly in this PR to support this change. On the other hand, the Sentry `SnubaEventStream` class (which is only used in tests) have been updated here: https://github.com/getsentry/sentry/pull/38877

Reply:
Updated: Both the old and new endpoints are kept, after this PR is merged, the https://github.com/getsentry/sentry/pull/38877 can be merged, finally the old eventstream endpoint can be safely removed.

Reply:
Does dataset actually need to be in here? I believe that inserting generally happens on a storage/entity and dataset is largely only for the query interface. I think that these eventstream endpoints should match the insert endpoints here (i.e. if you look at the write_to_entity function, there is no dataset passed, only an entity.

Reply:
Right, both new test `insert` and `eventstream` endpoints have been updated to only specify entity